client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,319 @@
+
+interface IndexedEventTarget extends EventTarget {
+  index: number;
+}
+
+interface TabLink extends HTMLAnchorElement {
+  index: number;
+}
+
+/**
+ * @example - creating a simple tabs interface
+ *<div class="w-tabs" data-controller="w-tabs" data-action="popstate@window->w-tabs#loadHistory" data-w-tabs-selected-value="" data-w-tabs-selected-class="animate-in">
+ *    <div class="w-tabs__list" role="tablist" data-w-tabs-target="list">
+ *        <a id="tab-label-tab-1" href="#tab-tab-1" class="w-tabs__tab" role="tab" tabindex="-1"
+ *           data-action="click->w-tabs#handleTabChange:prevent keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast" data-w-tabs-target="label">
+ *            Tab 1
+ *        </a>
+ *        <a id="tab-label-tab-2" href="#tab-tab-2" class="w-tabs__tab" role="tab"
+ *           data-action="click->w-tabs#handleTabChange:prevent keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast" data-w-tabs-target="label">
+ *            Tab 2
+ *        </a>
+ *    </div>
+ *    <div class="tab-content tab-content--comments-enabled">
+ *        <section id="tab-tab-1" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-1" data-w-tabs-target="panel">
+ *            tab-1
+ *        </section>
+ *        <section id="tab-tab-2" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-2" data-w-tabs-target="panel">
+ *            tab-2
+ *        </section>
+ *    </div>
+ *</div>
+ */
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['list', 'trigger', 'label', 'panel'];
+
+  static classes = ['selected'];
+
+  static values = {
+    transition: { default: 150, type: Number },
+    selected: { default: '', type: String },
+    syncURLHash: { default: false, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  declare readonly selectedClasses: string;
+
+  declare listTarget: HTMLDivElement;
+  declare triggerTargets: HTMLAnchorElement[];
+  declare labelTargets: HTMLAnchorElement[];
+  declare panelTargets: HTMLElement[];
+
+  declare syncURLHashValue: boolean;
+  declare selectedValue: string;
+  declare transitionValue: number;
+  declare animateValue: boolean;
+
+  connect() {
+    this.validate();
+
+    debounce(() => {
+      window.scrollTo({ top: 0, behavior: 'smooth' });
+    }, this.transitionValue * 2);
+
+    this.setAriaControls(this.labelTargets);
+    this.setTabLabelIndex();
+
+    const activeTab = this.labelTargets.find(
+      (button) =>
+        button.getAttribute('aria-selected') === 'true' ||
+        button.getAttribute('aria-controls') === this.selectedValue,
+    );
+
+    if (this.selectedClasses !== '' && activeTab) {
+      activeTab.setAttribute('aria-selected', 'true');
+      activeTab.removeAttribute('tabindex');
+    }
```

Comment:
*Classes will be an array of strings, so we need to check it's length.
https://stimulus.hotwired.dev/reference/css-classes

```suggestion
    if (this.selectedClasses.length && activeTab) {
      activeTab.setAttribute('aria-selected', 'true');
      activeTab.removeAttribute('tabindex');
    }
```